arch/xtensa/src/esp32s3/esp32s3_spi.h

```diff
@@ -0,0 +1,170 @@
+ *
+ *   1. Provide logic in esp32s3_board_initialize() to configure SPI chip
+ *      select pins.
+ *   2. Provide esp32s3_spi[0|1]_select() and esp32s3_spi[0|1]_status()
+ *      functions in your board-specific logic. These functions will perform
+ *      chip selection and status operations using GPIOs in the way your
+ *      board is configured.
+ *   3. If CONFIG_SPI_CMDDATA is defined in your NuttX configuration file,
+ *      then provide esp32s3_spi[0|1]_cmddata() functions in your
+ *      board-specific logic. These functions will perform cmd/data selection
+ *      operations using GPIOs in the way your board is configured.
+ *   4. Add a call to esp32s3_spibus_initialize() in your low level
+ *      application initialization logic.
+ *   5. The handle returned by esp32s3_spibus_initialize() may then be used
+ *      to bind the SPI driver to higher level logic (e.g., calling
+ *      mmcsd_spislotinitialize(), for example, will bind the SPI driver to
+ *      the SPI MMC/SD driver).
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ESP32S3_SPI2
+void esp32s3_spi2_select(struct spi_dev_s *dev, uint32_t devid,
+                         bool selected);
+uint8_t esp32s3_spi2_status(struct spi_dev_s *dev, uint32_t devid);
+int esp32s3_spi2_cmddata(struct spi_dev_s *dev,
+                         uint32_t devid,
+                         bool cmd);
+#endif
+
+/****************************************************************************
+ * Name: esp32s3_spibus_uninitialize
+ *
+ * Description:
+ *   Uninitialize an SPI bus.
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success.  Otherwise -1 (ERROR).
+ *
+ ****************************************************************************/
+
+int esp32s3_spibus_uninitialize(struct spi_dev_s *dev);
+
+/****************************************************************************
+ * Name: esp32s3_spislave_ctrlr_initialize
+ *
+ * Description:
+ *   Initialize the selected SPI Slave bus.
+ *
+ * Input Parameters:
+ *   port - Port number (for hardware that has multiple SPI Slave interfaces)
+ *
+ * Returned Value:
+ *   Valid SPI Slave controller structure reference on success;
+ *   NULL on failure.
+ *
+ ****************************************************************************/
+
+struct spi_slave_ctrlr_s *esp32s3_spislave_ctrlr_initialize(int port);
+
+/****************************************************************************
+ * Name: esp32s3_spislave_ctrlr_uninitialize
+ *
+ * Description:
+ *   Uninitialize an SPI Slave bus.
+ *
+ * Input Parameters:
+ *   ctrlr - SPI Slave controller interface instance
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success. Otherwise -1 (ERROR).
+ *
+ ****************************************************************************/
+
+int esp32s3_spislave_ctrlr_uninitialize(struct spi_slave_ctrlr_s *ctrlr);
+
```

Comment:
```suggestion
```
There is no SPI Slave support in this PR, so these functions should be removed from the driver interface.

Reply:
Functions are still present on the interface.

Reply:
Functions are still present on the interface.